common/src/main/java/me/lucko/luckperms/common/storage/dao/cassandra/CassandraDao.java

```diff
@@ -0,0 +1,500 @@
+    private PreparedStatement USER_UPDATE_PERMISSIONS;
+
+    private PreparedStatement GROUP_SELECT_ALL;
+    private PreparedStatement GROUP_SELECT;
+    private PreparedStatement GROUP_INSERT;
+    private PreparedStatement GROUP_DELETE;
+
+    private PreparedStatement TRACK_SELECT;
+    private PreparedStatement TRACK_SELECT_ALL;
+    private PreparedStatement TRACK_INSERT;
+    private PreparedStatement TRACK_DELETE;
+
+    private PreparedStatement UUID_TO_NAME_SELECT;
+    private PreparedStatement NAME_TO_UUID_SELECT;
+    private PreparedStatement UUID_TO_NAME_INSERT;
+    private PreparedStatement NAME_TO_UUID_INSERT;
+
+    public CassandraDao(LuckPermsPlugin plugin, CassandraConfig config) {
+        super(plugin, "Cassandra");
+        this.prefix = str -> str.replace("{prefix}", config.getPrefix());
+        this.config = config;
+    }
+
+    @Override
+    public void init() {
+        this.connectionManager = new CassandraConnectionManager(config);
+        Session session = connectionManager.getSession();
+        Cluster cluster = connectionManager.getCluster();
+        if (cluster.getMetadata().getKeyspace(config.getKeyspace()) == null) {
+            cluster.connect().execute("CREATE KEYSPACE IF NOT EXISTS " + config.getKeyspace() + " WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor':1};");
+            session.execute("USE " + config.getKeyspace());
+        } else {
+            session.execute("USE " + config.getKeyspace());
+        }
+        KeyspaceMetadata keyspace = session.getCluster().getMetadata().getKeyspace(config.getKeyspace());
+        UserType testTable = keyspace.getUserType(prefix.apply("permission"));
+        if(testTable == null) {
+            // create tables
+            String schemaFileName = "schema/cassandra.cql";
+            try (InputStream is = plugin.getResourceStream(schemaFileName)) {
+                if (is == null) {
+                    throw new Exception("Couldn't locate schema file for cassandra");
+                }
+
+                try (BufferedReader reader = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8))) {
+                    StringBuilder sb = new StringBuilder();
+                    String line;
+                    while ((line = reader.readLine()) != null) {
+                        if (line.startsWith("--") || line.startsWith("#")) continue;
+                        sb.append(line);
+                        if (line.endsWith(";")) {
+                            String result = prefix.apply(sb.toString());
+                            if(!result.isEmpty()) session.execute(result);
+                            sb = new StringBuilder();
+                        }
+                    }
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+                plugin.getLog().severe("Error occurred whilst initialising the database.");
+                shutdown();
```

Comment:
Remove the `#shutdown` call.

We allow failed initialisations so if the database comes back, the plugin will start to work.